cpp/src/parquet/encoding_test.cc

```diff
@@ -874,7 +874,7 @@ std::shared_ptr<::arrow::Array> EncodingAdHocTyped<FLBAType>::GetValues(int seed
 }
 
 using EncodingAdHocTypedCases =
-    ::testing::Types<BooleanType, Int32Type, Int64Type, FloatType, DoubleType, FLBAType>;
+    ::testing::Types<BooleanType, Int32Type, Int64Type, FloatType, DoubleType>;
```

Comment:
`FLBAType` was removed due to changes to `EncodingTraits<FLBAType>` above. Perhaps we could do something like:
```
  if constexpr (IsFixedSizeBinary::value) {
    using BuilderType = typename EncodingTraits<ParquetType>::BuilderType;
  } else {
    using BuilderType = typename EncodingTraits<ParquetType>::Accumulator;
  }
```
to get arround this.

Reply:
Just want to make clear why we modify the builder type? Maybe I've missed the context

Reply:
@wgtmac [proposed it](https://github.com/apache/arrow/pull/14341#discussion_r1256021740) to simplify `ArrowBinaryHelper` and there was no discussion yet. Would this be considered an external API change?

Reply:
Well... the thing is, the change brings pointless complication from the user's POV, isn't it? FixedSizeBinary arrays don't need to be broken down in multiple chunks because they have no limitation on data size.

If it's purely to ease templating, why don't you use a private templated helper? You even already have `ArrowBinaryHelper` which would perhaps be expanded or adapted.

Reply:
That's a good point. I reverted the `EncodingTraits<FLBAType>` change and expanded `ArrowBinaryHelper` to support `FLBA`.